File: VB/ChunkAndOptSummariesExample/ViewModel/ChunkAndOptSummariesViewModel.vb
Imports DevExpress.Xpf.ChunkList
Imports System.Collections.Generic

Namespace DataUpdatesExample

    Public Class ChunkAndOptSummariesViewModel
        Inherits UpdateViewModelBase

        Protected Overrides Sub ConfigureSettings()
            dataSourceSize = 1000000
            groupsCount = 100
            setAllRowValuesOnUpdate = False
            collectionUpdatesPerSecond = 200
            rowUpdatesPerSecond = 500
        End Sub

        Protected Overrides Function CreateSourceCollection() As IList(Of Device)
            Return New ChunkList(Of Device)(dataSourceSize)
        End Function
    End Class
End Namespace

File: CS/ChunkAndOptSummariesExample/ViewModel/ChunkAndOptSummariesViewModel.cs
using DevExpress.Xpf.ChunkList;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataUpdatesExample
{
    public class ChunkAndOptSummariesViewModel : UpdateViewModelBase
    {
        protected override void ConfigureSettings() {
            dataSourceSize = 1000000;
            groupsCount = 100;
            setAllRowValuesOnUpdate = false;
            collectionUpdatesPerSecond = 200;
            rowUpdatesPerSecond = 500;
        }
        protected override IList<Device> CreateSourceCollection() {
            return new ChunkList<Device>(dataSourceSize);
        }
    }
}
